docs/usage/getting-started/private-packages.md

```diff
@@ -324,6 +324,18 @@ npmRegistries:
 For each known NuGet registry, Renovate searches for `hostRules` with `hostType=nuget` and matching host.
 For those found, a command similar to the following is run: `dotnet nuget add source ${registryInfo.feedUrl} --configfile ${nugetConfigFile} --username ${username} --password ${password} --store-password-in-clear-text`
 
+password should be encoded to base64
```

Comment:
```suggestion
The `password` must be base64 encoded.
```

I think we mean _must_ instead of _should_ here. Or do we mean "We recommend you use base64 encoding, because other methods are wrong/dangerous?"

Are users allowed to use another encoding, or will that break things?

Reply:
according to rhys's comments its a must , but im not sure since there's no clear documentation, and that's what im trying to clarify here

Reply:
https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file#packagesourcecredentials
 
 in renovate we recommend encryption, but if you look at the link you will see
` If a non-encrypted password is passed for password the error message ["The parameter is incorrect" will occur](https://github.com/NuGet/Home/issues/3245).`

 its not clear if its base 64 or not, but it should be encrypted i guess,
 apparently there's another parameter called: ClearTextPassword
 for storing the actual password not encrypted.
 i'm not 100% sure how this is implemented so might need help here from @rarkins @viceice 
 
 if they cant answer this then i will have to investigate further

Reply:
i think we can remove this message for now, it's not necessary as in my example i wrote : Encrypted PAT token

Reply:
it needs to be plain text password or an API key, depends on the registry.

renovate encryption can be used, but never use base64 encoding.
that can maybe required by some custom registry, but not some i know / use.

Reply:
@viceice if you have experience here, would you please add some info to the documentation to make it clearer for users?
i've noticed we lack private packages handling documentation for most managers and it's costing us a lot of discussions and time

Reply:
I think there's maybe a mixup here regarding the npm password, where it's base64 encoded in `.npmrc` but we need it non-encoded in `hostRules`. I have no opinion on nuget passwords

Reply:
I'm simply use this host rule for my `proget` server and it works for `npm` and `nuget` datasources, `SERVICE_USER_PSW` is the plain text password passed via environment.

```js
  {
      matchHost: 'https://proget.domain.test/',
      username: 'service-user',
      password: process.env.SERVICE_USER_PSW,
    },
```

gitea nuget sample, they use plain password or PAT.
- https://docs.gitea.io/en-us/packages/nuget/#configuring-the-package-registry




> I think there's maybe a mixup here regarding the npm password, where it's base64 encoded in `.npmrc` but we need it non-encoded in `hostRules`. I have no opinion on nuget passwords

Yes, seemd to wrongly mixed up from npmrc

Reply:
in the discussion https://github.com/renovatebot/renovate/discussions/10998 they say they use PAT,
and it has 3 thumps up from other ppl too